apps/io-services-cms-webapp/src/utils/ingestion/ingestion-handlers.ts

```diff
@@ -22,15 +23,24 @@ export type IngestionResults<S> = IngestionError<S> | IngestionSuccess;
 const noAction = {};
 
 export const toEvents =
-  <S>(formatter: (item: S) => E.Either<Error, EventData>) =>
-  (items: readonly S[]): E.Either<Error, EventData[]> =>
-    pipe([...items], A.traverse(E.Applicative)(formatter));
+  <S, U extends S>(
+    formatter: RE.ReaderEither<U, Error, EventData>,
+    enricher: RTE.ReaderTaskEither<S, Error, U> = (item) => TE.right(item as U),
```

Comment:
suggestion: define a constant for the default enricher `(item) => TE.right(item as U)`